packages/library/src/commonMain/kotlin/io/realm/RealmModel.kt

```diff
@@ -16,10 +16,25 @@
 
 package io.realm
 
+import io.realm.internal.RealmModelInternal
+import io.realm.interop.RealmInterop
+
 // FIXME API Currently just adding these as extension methods as putting them directly into
 //  RealmModel would break compiler plugin. Reiterate along with
 //  https://github.com/realm/realm-kotlin/issues/83
 
 fun RealmObject.delete() {
     MutableRealm.delete(this)
 }
+
+var RealmObject.version: VersionId
```

Comment:
So I tried doing this as an extension variable `var RealmObject.version`. This seems to be a pretty bad situation because if the RealmObject itself has a version field, you have no way of accessing the extension property nor give it another name. The sample problem applies to functions though. 

I would suggest keeping it as is for now, and then create a ticket for going back to look at it. Ideally, we should move all the extension functions into the interface declaration itself, which would solve most of this (I hope).

If we, for whatever reason cannot do interface implementations, we should probably strongly consider renaming it to `realmVersion` or similar. 

What do you think?

Reply:
Just to clarify: The comment in the above FIXME about interface methods breaking the compiler plugin is not meaning that we cannot have them as interface methods with default implementation. We would just have to add all the interface methods as fake overrides on all user defined objects. It can probably be done in a generic way, so we don't have to do too many adjustments in the IR generation when adding new methods. But I guess the discussion is postponed to #219